source/slang/compiler.cpp

```diff
@@ -576,11 +575,14 @@ SlangResult dissassembleDXILUsingDXC(
         return 0;
     }
 
-    String dissassembleSPIRV(
+    SlangResult dissassembleSPIRV(
         CompileRequest*     slangRequest,
         void const*         data,
-        size_t              size)
+        size_t              size, 
+        String&             stringOut)
     {
+        stringOut = String();
```

Comment:
I personally like to name output or input/output parameters with an `out` or `io` prefix, respectively. I also don’t really like output reference parameters because they aren’t visible at the call site, but I’m okay with them when people find using pointer parameters less savory.

I’m also not sure about the convention here of initializing the output parameter at the start. I would usually expect a model where an output parameter is only written after all preconditions have been checked, so that in case of an error the output doesn’t get touched...

Reply:
I agree the reference parameter output is a bit problematic, can change to use pointer/s.

Reply:
The bug is that the dxbc disassembly is not written to the stringOut param. It's set on result and then thrown away. It originated when the SharedLibrary support was added.